pkg/storage/interface.go

```diff
@@ -14,10 +14,15 @@
 
 package storage
 
-import "sigs.k8s.io/metrics-server/pkg/api"
+import (
+	corev1 "k8s.io/api/core/v1"
+	apitypes "k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/metrics-server/pkg/api"
+)
 
 type Storage interface {
 	api.MetricsGetter
 	Store(batch *MetricsBatch)
+	DeleteRes(node *corev1.Node, podsRef []apitypes.NamespacedName)
```

Comment:
From an API perspective, `DeleteRes` isn't very intuitive. What would you think about adding two methods instead, one for removing NodeMetrics and another one for PodMetrics? They could be named:
- DiscardPods / DiscardNode

Also, we don't need to pass a pointer to the node here, just its name should be enough to delete it in the map.